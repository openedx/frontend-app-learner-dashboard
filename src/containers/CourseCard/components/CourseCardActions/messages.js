import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  beginCourse: {
    id: 'learner-dash.courseCard.actions.beginCourse',
    description: 'Course card begin-course button text',
    defaultMessage: 'Begin Course',
  },
  resume: {
    id: 'learner-dash.courseCard.actions.resume',
    description: 'Course card resume button text',
    defaultMessage: 'Resume',
  },
  viewCourse: {
    id: 'learner-dash.courseCard.actions.viewCourse',
    description: 'Course card view-course button text',
    defaultMessage: 'View Course',
  },
  selectSession: {
    id: 'learner-dash.courseCard.actions.selectSession',
    description: 'Course card select-session button text',
    defaultMessage: 'Select Session',
  },
});

export default messages;
