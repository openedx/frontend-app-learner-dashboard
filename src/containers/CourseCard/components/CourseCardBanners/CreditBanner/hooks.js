import { StrictDict } from '../../../../../utils';
import { reduxHooks } from '../../../../../hooks';

import ApprovedContent from './views/ApprovedContent';
import EligibleContent from './views/EligibleContent';
import MustRequestContent from './views/MustRequestContent';
import PendingContent from './views/PendingContent';
import RejectedContent from './views/RejectedContent';

export const statusComponents = StrictDict({
  pending: PendingContent,
  approved: ApprovedContent,
  rejected: RejectedContent,
});

export const useCreditBannerData = (cardId) => {
  const credit = reduxHooks.useCardCreditData(cardId);
  const { supportEmail } = reduxHooks.usePlatformSettingsData();
  if (!credit.isEligible) {
    return null;
  }

  const { error, purchased, requestStatus } = credit;
  let ContentComponent = EligibleContent;
  if (purchased) {
    if (requestStatus == null) {
      ContentComponent = MustRequestContent;
    } else if (Object.keys(statusComponents).includes(requestStatus)) {
      ContentComponent = statusComponents[requestStatus];
    }
    // Current behavior is to show Elligible State if unknown request status is returned
  }
  return {
    ContentComponent,
    error,
    supportEmail,
  };
};

export default {
  useCreditBannerData,
};
