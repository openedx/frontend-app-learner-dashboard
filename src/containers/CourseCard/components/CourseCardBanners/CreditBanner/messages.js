import { defineMessages } from '@openedx/frontend-base';

const messages = defineMessages({
  error: {
    id: 'learner-dash.courseCard.banners.credit.error',
    description: 'Error message for credit transaction with support email link',
    defaultMessage: 'An error occurred with this transaction.  For help, contact {supportEmailLink}.',
  },
  errorNoEmail: {
    id: 'learner-dash.courseCard.banners.credit.errorNoEmail',
    description: 'Error message for credit transaction without support email',
    defaultMessage: 'An error occurred with this transaction.',
  },
});

export default messages;
