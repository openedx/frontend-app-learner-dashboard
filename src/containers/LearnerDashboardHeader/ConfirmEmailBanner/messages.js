import { defineMessages } from '@openedx/frontend-base';

const messages = defineMessages({
  confirmNowButton: {
    id: 'leanerDashboard.confirmEmailBanner',
    description: 'Button for sending confirm email and open modal',
    defaultMessage: 'Confirm Now',
  },
  confirmEmailTextReminderBanner: {
    id: 'leanerDashboard.confirmEmailTextReminderBanner',
    description: 'Text for reminding user to confirm email',
    defaultMessage: 'Remember to confirm your email so that you can keep learning! {confirmNowButton}.',
  },
  verifiedConfirmEmailButton: {
    id: 'leanerDashboard.verifiedConfirmEmailButton',
    description: 'Button for verified confirming email',
    defaultMessage: 'I\'ve confirmed my email',
  },
  confirmEmailModalHeader: {
    id: 'leanerDashboard.confirmEmailModalHeader',
    description: 'title for confirming email modal',
    defaultMessage: 'Confirm your email',
  },
  confirmEmailModalBody: {
    id: 'leanerDashboard.confirmEmailModalBody',
    description: 'text hint for confirming email modal',
    defaultMessage: 'We\'ve sent you an email to verify your account. Please check your inbox and click on the big red button to confirm and keep learning.',
  },
  confirmEmailImageAlt: {
    id: 'leanerDashboard.confirmEmailImageAlt',
    description: 'text alt confirm email image',
    defaultMessage: 'confirm email background',
  },
});

export default messages;
