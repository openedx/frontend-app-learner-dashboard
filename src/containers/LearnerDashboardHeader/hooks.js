import React from 'react';
import { useIntl } from '@edx/frontend-platform/i18n';
import track from 'tracking';
import { StrictDict } from 'utils';
import { linkNames } from 'tracking/constants';

import getLearnerHeaderMenu from './LearnerDashboardMenu';

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavExplore,
});

export const useLearnerDashboardHeaderMenu = ({
  courseSearchUrl, authenticatedUser, exploreCoursesClick,
}) => {
  const { formatMessage } = useIntl();
  return getLearnerHeaderMenu(formatMessage, courseSearchUrl, authenticatedUser, exploreCoursesClick);
};

export default {
  findCoursesNavClicked,
  useLearnerDashboardHeaderMenu,
};
