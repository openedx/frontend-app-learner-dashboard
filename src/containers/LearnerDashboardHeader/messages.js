import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  dashboard: {
    id: 'learnerVariantDashboard.menu.dashboard.label',
    defaultMessage: 'Dashboard',
    description: 'The text for the user menu Dashboard navigation link.',
  },
  dashboardPersonal: {
    id: 'learnerVariantDashboard.menu.dashboardPersonal.label',
    defaultMessage: 'Personal',
    description: 'Link to personal dashboard in user menu',
  },
  dashboardSwitch: {
    id: 'learnerVariantDashboard.menu.dashboardSwitch.label',
    defaultMessage: 'SWITCH DASHBOARD',
    description: 'Switch Dashboard header in the user menu',
  },
  help: {
    id: 'learnerVariantDashboard.help.label',
    defaultMessage: 'Help',
    description: 'The text for the link to the Help Center',
  },
  profile: {
    id: 'learnerVariantDashboard.menu.profile.label',
    defaultMessage: 'Profile',
    description: 'The text for the user menu Profile navigation link.',
  },
  viewPrograms: {
    id: 'learnerVariantDashboard.menu.viewPrograms.label',
    defaultMessage: 'View Programs',
    description: 'The text for the user menu View Programs navigation link.',
  },
  account: {
    id: 'learnerVariantDashboard.menu.account.label',
    defaultMessage: 'Account',
    description: 'The text for the user menu Account navigation link.',
  },
  orderHistory: {
    id: 'learnerVariantDashboard.menu.orderHistory.label',
    defaultMessage: 'Order History',
    description: 'The text for the user menu Order History navigation link.',
  },
  logout: {
    id: 'learnerVariantDashboard.menu.logout.label',
    defaultMessage: 'Logout',
    description: 'The label for the user menu Logout action.',
  },
  course: {
    id: 'learnerVariantDashboard.course',
    defaultMessage: 'Courses',
    description: 'Header link for switching to dashboard page.',
  },
  program: {
    id: 'learnerVariantDashboard.program',
    defaultMessage: 'Programs',
    description: 'Header link for switching to program page.',
  },
  discoverNew: {
    id: 'learnerVariantDashboard.discoverNew',
    defaultMessage: 'Discover New',
    description: 'Header link for switching to discover page.',
  },
  logoAltText: {
    id: 'learnerVariantDashboard.logoAltText',
    defaultMessage: 'edX, Inc. Dashboard',
    description: 'Alt text for the edX logo.',
  },
  collapseMenuOpenAltText: {
    id: 'learnerVariantDashboard.collapseMenuOpenAltText',
    defaultMessage: 'Menu',
    description: 'Alt text for the collapse menu icon when the menu is open.',
  },
  collapseMenuClosedAltText: {
    id: 'learnerVariantDashboard.collapseMenuClosedAltText',
    defaultMessage: 'Close',
    description: 'Alt text for the collapse menu icon when the menu is closed.',
  },
  career: {
    id: 'leanerDashboard.menu.career.label',
    defaultMessage: 'Career',
    description: 'The text for the user menu Career navigation link.',
  },
  newAlert: {
    id: 'header.menu.new.label',
    defaultMessage: 'New',
    description: 'The text announcing that an item in the user menu is New',
  },
});

export default messages;
