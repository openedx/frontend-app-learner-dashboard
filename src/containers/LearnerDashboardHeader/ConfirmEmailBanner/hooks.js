import React from 'react';

import { StrictDict } from '@src/utils';
import { apiHooks, reduxHooks } from '@src/hooks';

import * as module from './hooks';

export const state = StrictDict({
  showPageBanner: (val) => React.useState(val), // eslint-disable-line
  showConfirmModal: (val) => React.useState(val), // eslint-disable-line
});

export const useConfirmEmailBannerData = () => {
  const { isNeeded } = reduxHooks.useEmailConfirmationData();
  const [showPageBanner, setShowPageBanner] = module.state.showPageBanner(isNeeded);
  const [showConfirmModal, setShowConfirmModal] = module.state.showConfirmModal(false);
  const closePageBanner = () => setShowPageBanner(false);
  const closeConfirmModal = () => setShowConfirmModal(false);
  const openConfirmModal = () => setShowConfirmModal(true);
  const sendConfirmEmail = apiHooks.useSendConfirmEmail();

  const openConfirmModalButtonClick = () => {
    sendConfirmEmail();
    openConfirmModal();
    closePageBanner();
  };

  const userConfirmEmailButtonClick = () => {
    closeConfirmModal();
    closePageBanner();
  };
  return {
    isNeeded,
    showPageBanner,
    closePageBanner,
    showConfirmModal,
    closeConfirmModal,
    openConfirmModalButtonClick,
    userConfirmEmailButtonClick,
  };
};

export default useConfirmEmailBannerData;
