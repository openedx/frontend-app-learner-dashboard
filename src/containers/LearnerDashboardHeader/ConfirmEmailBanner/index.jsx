/* eslint-disable max-len */
import React from 'react';
import {
  Button,
  Image,
  MarketingModal,
  ModalDialog,
  PageBanner,
} from '@openedx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import confirmEmailSVG from 'assets/confirm-email.svg';
import messages from './messages';
import './ConfirmEmailBanner.scss';
import useConfirmEmailBannerData from './hooks';

export const ConfirmEmailBanner = () => {
  const {
    isNeeded,
    showConfirmModal,
    showPageBanner,
    closePageBanner,
    closeConfirmModal,
    openConfirmModalButtonClick,
    userConfirmEmailButtonClick,
  } = useConfirmEmailBannerData();
  const { formatMessage } = useIntl();

  if (!isNeeded) { return null; }

  return (
    <>
      <PageBanner show={showPageBanner} dismissible onDismiss={closePageBanner}>
        {formatMessage(messages.confirmEmailTextReminderBanner, {
          confirmNowButton: (
            <Button
              className="confirm-email-now-button"
              variant="link"
              size="inline"
              onClick={openConfirmModalButtonClick}
            >
              {formatMessage(messages.confirmNowButton)}
            </Button>
          ),
        })}
      </PageBanner>
      <MarketingModal
        title=""
        isOpen={showConfirmModal}
        onClose={closeConfirmModal}
        hasCloseButton={false}
        heroNode={(
          <ModalDialog.Hero className="bg-gray-300">
            <Image
              className="m-auto"
              src={confirmEmailSVG}
              alt={formatMessage(messages.confirmEmailImageAlt)}
            />
          </ModalDialog.Hero>
        )}
        footerNode={(
          <Button className="mx-auto my-3" variant="danger" onClick={userConfirmEmailButtonClick}>
            {formatMessage(messages.verifiedConfirmEmailButton)}
          </Button>
        )}
      >
        <h2 className="text-center p-3 h1">{formatMessage(messages.confirmEmailModalHeader)}</h2>
        <p className="text-center">{formatMessage(messages.confirmEmailModalBody)}</p>
      </MarketingModal>
    </>
  );
};
ConfirmEmailBanner.propTypes = {};

export default ConfirmEmailBanner;
