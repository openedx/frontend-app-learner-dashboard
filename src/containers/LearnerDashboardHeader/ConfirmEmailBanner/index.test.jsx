import { render, screen } from '@testing-library/react';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import hooks from './hooks';
import ConfirmEmailBanner from '.';
import messages from './messages';

jest.mock('./hooks', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.unmock('@openedx/paragon');
jest.unmock('@edx/frontend-platform/i18n');
jest.unmock('react');

const hookProps = {
  isNeeded: true,
  showPageBanner: jest.fn().mockName('showPageBanner'),
  closePageBanner: jest.fn().mockName('closePageBanner'),
  showConfirmModal: jest.fn().mockName('showConfirmModal'),
  closeConfirmModal: jest.fn().mockName('closeConfirmModal'),
  openConfirmModalButtonClick: jest.fn().mockName('openConfirmModalButtonClick'),
  userConfirmEmailButtonClick: jest.fn().mockName('userConfirmEmailButtonClick'),
};

describe('ConfirmEmailBanner', () => {
  describe('render', () => {
    it('do not show on already verified', () => {
      hooks.mockReturnValueOnce({ ...hookProps, isNeeded: false });
      render(<IntlProvider locale="en"><ConfirmEmailBanner /></IntlProvider>);
      const banner = screen.queryByRole('alert');
      expect(banner).toBeNull();
    });
    describe('On unverified', () => {
      beforeEach(() => {
        hooks.mockReturnValueOnce({ ...hookProps });
        render(<IntlProvider locale="en"><ConfirmEmailBanner /></IntlProvider>);
      });
      it('show banner', () => {
        const banner = screen.getByRole('alert');
        expect(banner).toContainHTML('Remember to confirm');
      });
      it('show confirm now button', () => {
        const confirmButton = screen.getByRole('button', { name: messages.confirmNowButton.defaultMessage });
        expect(confirmButton).toBeInTheDocument();
      });
      it('show modal', () => {
        const modal = screen.getByRole('dialog');
        expect(modal).toBeInTheDocument();
      });
      it('show modal header and body', () => {
        const modalHeader = screen.getByText(messages.confirmEmailModalHeader.defaultMessage);
        expect(modalHeader).toBeInTheDocument();
        const modalBody = screen.getByText(messages.confirmEmailModalBody.defaultMessage);
        expect(modalBody).toBeInTheDocument();
      });
      it('show confirm image', () => {
        const confirmButton = screen.getByRole('img', { name: messages.confirmEmailImageAlt.defaultMessage });
        expect(confirmButton).toBeInTheDocument();
      });
      it('show confirm email button', () => {
        const confirmButton = screen.getByRole('button', { name: messages.verifiedConfirmEmailButton.defaultMessage });
        expect(confirmButton).toBeInTheDocument();
      });
    });
  });
});
