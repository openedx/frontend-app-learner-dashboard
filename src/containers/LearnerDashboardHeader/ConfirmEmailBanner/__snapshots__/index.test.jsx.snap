// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailBanner snapshot Show on unverified 1`] = `
<Fragment>
  <PageBanner
    dismissible={true}
    onDismiss={[MockFunction closePageBanner]}
    show={[MockFunction showPageBanner]}
  >
    <format-message-function
      message={
        {
          "defaultMessage": "Remember to confirm your email so that you can keep learning! {confirmNowButton}.",
          "description": "Text for reminding user to confirm email",
          "id": "leanerDashboard.confirmEmailTextReminderBanner",
        }
      }
      values={
        {
          "confirmNowButton": <Button
            className="confirm-email-now-button"
            onClick={[MockFunction openConfirmModalButtonClick]}
            size="inline"
            variant="link"
          >
            Confirm Now
          </Button>,
        }
      }
    />
  </PageBanner>
  <MarketingModal
    footerNode={
      <Button
        className="mx-auto my-3"
        onClick={[MockFunction userConfirmEmailButtonClick]}
        variant="danger"
      >
        I've confirmed my email
      </Button>
    }
    hasCloseButton={false}
    heroNode={
      <ModalDialog.Hero
        className="bg-gray-300"
      >
        <Image
          alt="confirm email background"
          className="m-auto"
          src="icon/mock/path"
        />
      </ModalDialog.Hero>
    }
    isOpen={[MockFunction showConfirmModal]}
    onClose={[MockFunction closeConfirmModal]}
    title=""
  >
    <h2
      className="text-center p-3 h1"
    >
      Confirm your email
    </h2>
    <p
      className="text-center"
    >
      We've sent you an email to verify your account. Please check your inbox and click on the big red button to confirm and keep learning.
    </p>
  </MarketingModal>
</Fragment>
`;

exports[`ConfirmEmailBanner snapshot do not show on already verified 1`] = `null`;
